File: sprint02/JMS/BOJ_2606.cpp
#include <iostream>

using namespace std;

int N, M, cnt = 0;
int arr[100][100] = { 0 };
int visited[100] = { 0 };

void DFS(int vertex) {
	cnt++;
	visited[vertex] = 1;

	for (int i = 0; i < N; i++) {
		if (arr[vertex][i] && !visited[i])
			DFS(i);
	}
}

void Input() {
	cin >> N >> M;

	for (int i = 0; i < M; i++) {
		int vertex1, vertex2;

		cin >> vertex1 >> vertex2;
		arr[vertex1 - 1][vertex2 - 1] = 1;
		arr[vertex2 - 1][vertex1 - 1] = 1;
	}
}

int main() {
	Input();
	DFS(0);
	cout << cnt - 1;
}

File: sprint02/KSH/BOJ_2606.py

from collections import deque

# bfs
def bfs(lst):

    queue = deque()
    visited = []
    queue.extend(lst[1])
    while queue:
        l = queue.popleft()
        if l not in visited:
            visited.append(l)
            queue.extend(lst[l])
    
    return (len(visited) - 1)


def dfs(idx, lst, visited):
    global cnt

    for i in lst[idx]:
        if i not in visited:
            visited.append(i)
            dfs(i, lst, visited)
            cnt += 1

if __name__ == "__main__":
    n = int(input())
    m = int(input())
    lst = [[] for i in range(n + 1)]
    visited = [0] * (n + 1)

    for i in range(m):
        a, b = map(int, input().split())
        lst[a].append(b)
        lst[b].append(a)

    # global cnt
    cnt = 0

    visited = []
    dfs(1, lst, visited)

    print(cnt - 1)

